### ISSUE TITLE
deleting images improperly added to Filemaker in order to run QR upload

### ISSUE THREAD

**Issue by lbestock** [urapadmin/kiosk#753]
This is just a ticket to remind Laurel to watch out for this cycle of syncrhonization because Luiza's process around QR code photos might cause interest.

She took QR code photos. She then however put them directly into Twix, using airdrop to get them off her phone.

She uploaded. I imported but had not yet syncrhonized when she made me aware that she'd done this.

She went back into the file and deleted the images, then uploaded again.

Two things I'm interested in (which Lutz could answer immediately but we're trying not to overwhelm him in the daily grind of the season!): since they're still in the gallery, not forgotten entirely, will they be accepted as uploads at all? (Can I delete things from the gallery as an admin?) And did the uploading again work? It said so on Luiza's end. Kiosk did not tell me to import after that, though, it still said "ready to syncrhonize". I was able to ask it to import again. The process manager told me about my own importing, but had not mentioned to me Luiza's upload. 

Here goes....

**Comment by lbestock** [comment]
Failed spectacularly! Good. Also from Sam's end because I did the simple synchronization and he was testing images that were already in there. But enough has gone on that I want to talk to Lutz before doing anything else.

It is an interesting test case because I think it is very likely to have real-world correlates, both on the duplicate images and on the deleted images front. The instinct to erase one's mistakes is really big. I note that this was also an issue in paper recording and one reason I was taught always to record in pen; even though pencil is more archival, my directors wanted no possibility of people trying to erase stuff. So we need to write in the users manual instructions for what to do if an excavator deletes things and that screws stuff up, whether the mechanism for upscrewing is deletion or the upload import frenzy.

**Comment by urapadmin** [comment]
It is rather chaotic here. I try to sort things out:

First of all:
That she uploaded again and the system did not tell you to import again could be seen as a glitch. After an upload there must be an import (you could have chosen it from "more options"), because it is the import that opens the file and imports the workstation data into the server. If you did not import again you will just synchronize what had been imported before.

1. she would indeed have to delete them in the gallery to prevent them from entering the server database.  If she did not do that, which I suspect, since that would even need admin privileges, they will enter the file repository as candids (because Luiza cut the connection to a context). 
2. yes, admins can deleted images. Unlike in the kiosk file repository deleting an image in the gallery does NOT delete the reference to the image. That needs to be done first.
3. Once synchronized, the images would prevent a file import since the imported images would be duplicates. 

To complicate things: The import step, although repeatable, isn't entirely repeatable. The requirement is already somewhere in an older ticket but, you know ... 
It would be a bit technical to explain here the difference between the first and the second import, and I would have trouble to come up with all the possible consequences from importing again, but different data. To be on the safest side I should make it impossible. Uploading can be repeated as often as necessary, but once imported, synchronization is the only possible next step. 

So how would one deal with this particular issue (an issue I do not expect to happen in the field ever)? Easy. You synchronize Luiza's images. If they are connected to a context then, well, there is nothing left to do, the QR code was just entirely unnecessary. 
If (like in this case) the context has been removed, all you need to do is remove the images themselves in the file repository. And then import the qr-code photos through the file import again.

**Comment by lbestock** [comment]
I HAD imported it again - I figured as much. But since that is called a glitch I will make another ticket.

Much of the best practice here I anticipated. But I did definitely want to see what happens and how confusing it is to a director if an excavator does this. I think it is relevant for the users manual, a whole section on how to delete.

But first we need more information about how there came to be duplicates in the first place. Luiza - describe your actions!

**Comment by luizaogs** [comment]
Ok here's a list of the broken images with attached explanations (or, rather, more questions). These are in order of the broken files in the file repository if one sorts them from latest to earliest in case you want to follow up on specific cases (hopefully this makes them easier to find): 

1 — LA test, so in Playground, dated to today. This is a weird one. I was trying to figure out what the "insert qr code substitute" thing did earlier, which is why I realized I had dealt with the QR codes incorrectly in the first place. But anyway, this isn't a broken image -- it just says "substituting" on there (I guess I forgot to delete it before re-uploading the whole thing to Kiosk):

![image](https://user-images.githubusercontent.com/67970402/87883919-1c6dbd80-c9e1-11ea-8afd-5f09ca949e10.png)

2 — SB purposefully duplicate

3 & 4 — these are the weird ones. I noticed that I had in fact uploaded it twice this morning when I was going through the gallery but then I deleted it before importing and now there is only one of them in there. But the one in the gallery (see first screenshot) does not say that it is also under the context relations for LA-011 and LA-012, which it should be (and it is in fact there from looking at the contexts themselves — see screenshots 2-4 — so I am not sure what is going on here). Hmm though now I’m looking at the file repository and there’s one in there without context of the same image (see screenshot 5). Could the issue be related with the fact that I deleted one of them but it is still in the gallery?

![image](https://user-images.githubusercontent.com/67970402/87883953-4fb04c80-c9e1-11ea-8018-6d76c94cafa6.png)

![image](https://user-images.githubusercontent.com/67970402/87883955-52ab3d00-c9e1-11ea-983e-ec8d4f8fc7a4.png)

![image](https://user-images.githubusercontent.com/67970402/87883957-55a62d80-c9e1-11ea-8880-2167595e8f46.png)

![image](https://user-images.githubusercontent.com/67970402/87883958-58a11e00-c9e1-11ea-9bf9-24753d1f1ee7.png)

![image](https://user-images.githubusercontent.com/67970402/87883984-838b7200-c9e1-11ea-8a07-fee91995e86b.png)

5-9 — these are all Playground files. I totally forgot I needed to worry about Playground files. I’m sorry, I seem to be on a role this week.

10, 11-16 — these are without context so I assume my QR photos, but in that case I have no idea why they would be duplicate. I would not have uploaded them more than once because most are collected material QR photos. Not at all sure what to do with these:

![image](https://user-images.githubusercontent.com/67970402/87884011-a9187b80-c9e1-11ea-83a8-3f428a2a1da5.png)

17 & 18 — Nuri duplicate photos so not sure what went on there

19 & 20 — again duplicates with no context. See explanation for images 10, 11-16.

In better, less confusing news, the QR images that are actually in there and not broken are indeed context-less, so if Laurel deletes them I should be able to upload them as they should have been the first time.

**Comment by urapadmin** [comment]
This will take me quite some time to address in detail ...

**Comment by luizaogs** [comment]
I'm sorry!!!

**Comment by urapadmin** [comment]
don't get me wrong, this is invaluable. That's why it will take a bit more time. I have saved the workstation you uploaded last on Beset so that you guys can carry on.

**Comment by urapadmin** [comment]
I might call you Sherlock from now on.

**Comment by luizaogs** [comment]
Well that is quite soothing for my damaged ego.

**Comment by urapadmin** [comment]
1 — LA test, so in Playground, dated to today. This is a weird one. I was trying to figure out what the "insert qr code substitute" thing did earlier, which is why I realized I had dealt with the QR codes incorrectly in the first place. But anyway, this isn't a broken image -- it just says "substituting" on there (I guess I forgot to delete it before re-uploading the whole thing to Kiosk):

![image](https://user-images.githubusercontent.com/67970402/87883919-1c6dbd80-c9e1-11ea-8afd-5f09ca949e10.png)

see #760 I can only imagine that you got those error messages and dismissed them? At least the "substituting..." is only an intermediary state and is usually finished with a qr-code in the image field. Nevermind, the whole thing is outdated and does not work properly anyhow, as #760 shows you.

**Comment by urapadmin** [comment]
> 3 & 4 — these are the weird ones. I noticed that I had in fact uploaded it twice this morning when I was going through the gallery but then I deleted it before uploading and now there is only one of them in there. But the one in the gallery (see first screenshot) does not say that it is also under the context relations for LA-011 and LA-012, which it should be (and it is in fact there from looking at the contexts themselves — see screenshots 2-4 — so I am not sure what is going on here). Hmm though now I’m looking at the file repository and there’s one in there without context of the same image (see screenshot 5). Could the issue be related to the fact that I deleted one of them but it is still in the gallery? The file repository in fact only seems to have the one that was deleted from the gallery, which is why it is there with no context and the other two appear broken.
> 
> ![image](https://user-images.githubusercontent.com/67970402/87884107-4f648100-c9e2-11ea-93d4-0ae502af163e.png)
> 
> ![image](https://user-images.githubusercontent.com/67970402/87883955-52ab3d00-c9e1-11ea-983e-ec8d4f8fc7a4.png)
> 
> ![image](https://user-images.githubusercontent.com/67970402/87883957-55a62d80-c9e1-11ea-8880-2167595e8f46.png)
> 
> ![image](https://user-images.githubusercontent.com/67970402/87883958-58a11e00-c9e1-11ea-9bf9-24753d1f1ee7.png)
> 
> ![image](https://user-images.githubusercontent.com/67970402/87883984-838b7200-c9e1-11ea-8a07-fee91995e86b.png)
> 

A bunch of things together here:
a) you were looking for the images in the gallery using "LA" and "strict". Consequently you only find images that are explicitly told to react to the term LA. That isn't the case for locus relation images because we decided that we don't want all the locus relation images to show up when somebody is searching for images of unit LA. That is why you don't see them in the gallery. 
b) 
![image](https://user-images.githubusercontent.com/38838314/87915570-a2861480-ca72-11ea-9518-32e787b9360f.png)
Here you can see the missed images because I searched for LA-010. And you see that we have indeed ONE duplicate. The image is attached to LA-010, LA-011 and LA-012.

**Comment by urapadmin** [comment]
> 10, 11-16 — these are without context so I assume my QR photos, but in that case I have no idea why they would be duplicate. I would not have uploaded them more than once because most are collected material QR photos. Not at all sure what to do with these:
> 
> ![image](https://user-images.githubusercontent.com/67970402/87884011-a9187b80-c9e1-11ea-83a8-3f428a2a1da5.png)
>

**Comment by urapadmin** [comment]
for the future (and please communicate this to everybody):

1 & 2 and such references don't tell me much because that's highly dependent on you search terms and what not. So if you want to reference an image, this is the best way to do it: Open it in the file repository and copy the uid in the upper right corner:
![image](https://user-images.githubusercontent.com/38838314/87917571-7b7d1200-ca75-11ea-9ef7-a31d41a895b7.png)

You can mark it with the mouse and copy it to the clipboard (You all know CTRL+C, I suspect). You don't have to type that number. 
And by the way: You can paste such a number in the file respository in "description" and the system searches for exactly that image.

If I have the uid I also don't need an image, which relieves pressure on Github and the global climate. I am still surprised that they let us upload tons of photos free of charge.

**Comment by urapadmin** [comment]
so far I have not seen anything like a real error here. Of course, the duplicate file issue is an issue but there are already tickets for future improvements. 

Please check if I have overlooked something. e.g. I have not looked at what you called "Nuri's images" because I was not sure which ones were meant exactly. 

If you think the major issues are covered and explained here, please don't forget to close this issue.

**Comment by luizaogs** [comment]
A couple of follow up points: 

For the LA-010, -011, -012 one -- I guess definitely something to keep in mind for the manual, the strict + locus relations thing. I am still not sure about this just because as stated in #743 the carrying over of images in locus relations seems to have been weird precisely when I was dealing with these so something might have happened there (though it is DEFINITELY possible that I just put it in twice, but I've been looking out for that so would have to have been distracted). The error might have arisen from something to do with inputting locus relations images, so something to check next time I put in data, I think.

The duplicates from LA test bring up an interesting question (for me, at least, not sure if you have discussed this before and come to different conclusions): those are all from Playground, when I was playing with the narrative link thing and so putting things in and deleting them repeatedly. Truly my fault for not realizing that Playground stuff is actually also uploaded to file repository, but this at the very least brings up important stuff for the manual, too, I think: if you work on stuff in Playground you will get the data regardless. On that note, I wonder if it is possible/worth it to somehow separate the two in file repository? I know that if you search by context you will obviously not get those, but if you're fetching them all it could be helpful to have the option to fetch just the stuff from the actual site, since I imagine the Playground data is usually what we have here -- a mess. The reason I did it in Playground was so that it WOULDN'T interfere with data or synchronization or anything like that, but then it did anyway. Also worth thinking about whether the Playground data should be uploaded at all, if it is even possible to not do so? Is there ever a scenario where that data is desired? 

Nuri's images are 83f33318-d122-4b37-abfe-8f3b6150afe7 and b23577f0-33f6-40a1-87db-7006d3dfbc65 and I note that those are also context relation images as the ones above so definitely something to check next time. I suspect something is going on there, especially since in #743 I seemed to think that images from the camera roll were carrying over (and they were, but I don't know if by doing so they could possibly be duplicating themselves?).

**Comment by urapadmin** [comment]
as for the playground: We do scenario testing to get exactly the somewhat chaotic response we are getting here. It is something one never gets in more structured, clinical testing. So, all is good, Luiza. Don't even try to avoid errors. We want them (exhausting as they often are, indeed, given the mounting tickets, but that something I have to deal with)

**Comment by luizaogs** [comment]
I did not think I had inserted it twice, but I must have done it accidentally. I'll just triple check next time to make sure I don't do it again -- not sure how else we would look into it since I don't think I can really reconstruct my steps for this specific instance.

**Comment by urapadmin** [comment]
Meanwhile I was able to look at the two images from Nuri. They are in fact the same image. So it has been inserted three times and two were rejected as duplicates:
![image](https://user-images.githubusercontent.com/38838314/87991759-2843a780-cae7-11ea-80f8-ad4a403ea5ef.png)
It seems it was first inserted as a dayplan on the 15. and then twice as a locus relation.

An idea: Copy & Paste creates a new image! It might be tempting to create one relation and add an image and then copy and paste that same image into the next relation instead of using the gallery again.  I don't think that happened here, but it just occurred to me as a potential source of such duplicates particularly while we are testing:
![image](https://user-images.githubusercontent.com/38838314/87991965-7e184f80-cae7-11ea-817d-f980be7d8f6c.png)

So let's keep an eye on those locus relation photos and perhaps make them a single image container test (instead of in the scenario where so many things happen over days). @SamKimball-BrownU, you are the expert on image container testing.

**Comment by luizaogs** [comment]
A bit of a weird issue that I'm not sure how to deal with; it doesn't seem likely to have a real world correlate but who knows. As I mentioned above, that one damn photograph in the context relations LA-010, -011, -012, etc. is only present in the gallery currently without a context. I can't access the context-less images in the gallery, as far as I can see? I see the broken image in there but that doesn't help me. In this case, what is the best way to replace the broken image with the actual image in the database? Delete the context-less and broken stuff in the gallery (which has to be done by an admin, correct?) so that I can re-upload it?

I commented here because this is clearly very related to all the above, but if I should open a new ticket just let me know.

**Comment by urapadmin** [comment]
please tell me which photograph. In fact you should not even see context-less images in filemaker. If images are not connected to recording data they don't get transported from kiosk to the recording. (We have tooons of candids in the Urap database that would make filemaker explode). So something is wrong about this particular image.

**Comment by luizaogs** [comment]
Woops sorry, getting my terms confused. It's visible in the file repository, not the gallery (2c33428a-77b3-48df-8793-6819dcf8d182), and that's the whole issue because I can't insert it back from the gallery in this case -- the two versions of it in there are already broken.

**Comment by urapadmin** [comment]
I see. That's kinda circular. The image in the file repository is not connected to a context and so does not get into filemaker but it keeps the same image one might add to filemaker from entering the file repository. 
Good we have kiosk 0.8.7 now. You can got to that image 2c33428a-77b3-48df-8793-6819dcf8d182 and assign it to a context there. With the next synchronization it will be available in the gallery again and then you could go to all the other places where it belongs (and where you should have empty image containers) and insert it there from the gallery.

There is in fact only one real solution to the whole duplicate image horror, which will happen on all projects over and over again and that is #441.

From my point of view we could close this issue here. If you rectify the issue in the data is up to you - in the end it is all test data - but technically everything is working as defined.

**Comment by luizaogs** [comment]
Got it, context assigned, thanks! 

I mentioned the questions I had about Playground and how it might get in the way of synchronization etc. in a real field scenario at the meeting yesterday and it sounded like Laurel might want more discussion of those, but since the users manual is tagged here I guess we can close it?

**Comment by lbestock** [comment]
We can indeed. Also with #441 (I didn't know about that until this reference - wow) images will cease to be a problem in this way at all, and so leaving duplicates in the Playground wouldn't cause synchronization problems. You are to date our only user to ever seriously use the Playground :) It is appreciated. We'll keep our eye on it but I have my doubts that it would ever get big enough to really be a weight, and since structurally it is just another site, if ever there are Playground problems they are whole system problems so we WANT to run into them.
